external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -3393,12 +3421,15 @@ fn resolve_exp_dotted(
 
     if let Some(loc) = autocomplete_last {
         assert!(context.env.ide_mode());
-        debug_print!(context.debug.autocomplete_resolution, ("computing autocomplete" => result; dbg));
+        debug_print!(context.debug.autocomplete_resolution, ("computing unresolved dot autocomplete" => result; dbg));
         ide_report_autocomplete(context, &loc, &edotted_ty);
-        result
-    } else {
-        result
     }
+    if context.env.ide_mode() {
+        if let Some(mdot_loc) = method_dot_loc {
+            ide_report_autocomplete(context, &mdot_loc, &edotted_ty);
+        }
+    }
+    result
```

Comment:
This check is redundant.

```suggestion
    if let Some(mdot_loc) = method_dot_loc {
        ide_report_autocomplete(context, &mdot_loc, &edotted_ty);
    }
    result
```

Reply:
It wasn't 100% redundant as there were places where `method_dot_loc` argument was set to an actual location that were not guarded by the `ide_mode` flag, but it is now after some minor additional changes to `method_call` and `macro_method_call` functions